grpc/pb-marshalling.go

```diff
@@ -234,3 +237,123 @@ func argsToPerformValidationRequest(domain string, challenge core.Challenge, aut
 	}, nil
 
 }
+
+func registrationToPB(reg core.Registration) (*rapb.Registration, error) {
+	keyBytes, err := reg.Key.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	ipBytes, err := reg.InitialIP.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	createdAt := reg.CreatedAt.UnixNano()
+	status := string(reg.Status)
+	var contacts []string
+	// Since the default value of rapb.Registration.Contact is a slice
+	// we need a indicator as to if the value is actually important on
+	// the other side (pb -> reg).
+	contactsPresent := reg.Contact != nil
+	if reg.Contact != nil {
+		contacts = *reg.Contact
+	}
+	return &rapb.Registration{
+		Id:              &reg.ID,
+		Key:             keyBytes,
+		Contact:         contacts,
+		ContactsPresent: &contactsPresent,
+		Agreement:       &reg.Agreement,
+		InitialIP:       ipBytes,
+		CreatedAt:       &createdAt,
+		Status:          &status,
+	}, nil
+}
+
+func pbToRegistration(pb *rapb.Registration) (core.Registration, error) {
+	var key jose.JsonWebKey
+	err := key.UnmarshalJSON(pb.Key)
+	if err != nil {
+		return core.Registration{}, err
+	}
+	var initialIP net.IP
+	err = initialIP.UnmarshalText(pb.InitialIP)
+	if err != nil {
+		return core.Registration{}, err
+	}
+	var contacts *[]string
+	if *pb.ContactsPresent {
+		if len(pb.Contact) != 0 {
+			contacts = &pb.Contact
+		} else {
+			// When gRPC creates an empty slice it is actually a nil slice, since
```

Comment:
s/,/./. Also this comment should specifically describe how this is a dependency for deleting contacts.